packages/firestore/src/core/sync_engine.ts

```diff
@@ -653,11 +660,37 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
   // PORTING NOTE: Multi-tab only
   applyPrimaryState(isPrimary: boolean): Promise<void> {
     this.isPrimary = isPrimary;
-    return Promise.resolve();
+    if (this.isPrimary && this.networkEnabled) {
+      return this.remoteStore.enableNetwork();
+    } else {
+      return this.remoteStore.disableNetwork();
+    }
   }
 
   // PORTING NOTE: Multi-tab only
   getActiveClients(): Promise<ClientId[]> {
     return this.localStore.getActiveClients();
   }
+
+  async enableNetwork(): Promise<void> {
+    this.networkEnabled = true;
+    if (this.isPrimary) {
+      return this.remoteStore.enableNetwork();
+    }
+  }
+
+  async disableNetwork(): Promise<void> {
+    // TODO(multi-tab): Release primary lease
```

Comment:
Is it multi-tab or multitab?  Not sure if the code is already inconsistent, but you're inconsistent in this PR...